async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+			
+			JSONObject jsonObject = new JSONObject(receivemessage);
+	
+	
+			transactionId = jsonObject.getString("transactionid");
+			
+			destinationUri = jsonObject.getString("destinationuri");
+			
+			destinationHttpMethod = jsonObject.getString("destinationhttpmethod");
+			
+			asyncId = jsonObject.getString("asyncid");
+			
+			callbackUri = jsonObject.getString("callbackuri");
+			
+			retryTime = Integer.parseInt(jsonObject.getString("executiontimes").toString()) + 1;
+			
+	
+			retrymessage.append("{");
+			retrymessage.append("\"transactionid\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(transactionId);
+			retrymessage.append("\"");
+			retrymessage.append(",");
+			retrymessage.append("\"destinationuri\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(destinationUri);
+			retrymessage.append("\"");
+			retrymessage.append(",");
+			retrymessage.append("\"destinationhttpmethod\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(destinationHttpMethod);
+			retrymessage.append("\"");
+	
+			if(receivemessage.contains("destinationrequestheader")) {
+				destinatioRrequestHeader = jsonObject.getString("destinationrequestheader");
+				
+				retrymessage.append(",");
+				retrymessage.append("\"destinationrequestheader\"");
+				retrymessage.append(":");
+				retrymessage.append("\"");
+				retrymessage.append(destinatioRrequestHeader);
+				retrymessage.append("\"");
+			}
+			
+	
+			if(receivemessage.contains("destinationrequestbody")) {
+				destinationRequestBody = jsonObject.getString("destinationrequestbody");
+				
+				retrymessage.append(",");
+				retrymessage.append("\"destinationrequestbody\"");
+				retrymessage.append(":");
+				retrymessage.append("\"");
+				retrymessage.append(destinationRequestBody);
+				retrymessage.append("\"");
+			}
+			
+			retrymessage.append(",");
+			retrymessage.append("\"callbackuri\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(callbackUri);
+			retrymessage.append("\"");
+			retrymessage.append(",");
+			retrymessage.append("\"asyncid\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(asyncId);
+			retrymessage.append("\"");
+	
+			retrymessage.append(",");
+			retrymessage.append("\"executiontimes\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(Integer.toString(retryTime));
+			retrymessage.append("\"");
+			retrymessage.append("}");
+			
+		} catch(Exception e) {
+			e.printStackTrace();
+			return GeneralErrorRes(ErrorResponseEnum.E500);
+		} 
+				
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	}
+	
+	private Response CreateSendMessage(String receivemessage, StringBuilder sendmessage, StringBuilder destinationStatus, StringBuilder destinationResponseBody) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 59 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.